### PR TITLE
Extra import to summon a Timer[Task] instance

### LINKED ISSUE #694: Make integration with Cats Effect more friendly
Right now, the user requires an `implicit runtime: Runtime[R]` for their chosen environment `R` in order to summon the cats-effect instances.

It would be nice to provide this automatically. Possibly we can change `App`

```scala
trait App { 
  implicit runtime: Runtime[Environment] = this
  ...
}
```

Then at least, this makes `App` ready-to-go for Cats Effect. The drawback is that this implicit will make using custom environments harder. Indeed, `App` itself hard codes the `Environment` to be the core set of effects in ZIO—perhaps this should change?

There's still the issue of "what type do I use for cats-effect", which unfortunately varies depending on whether or not the user requires `Timer`.

Possibly in the cats-interop package, we could provide:

```scala
type TimerTask[A] = TaskR[Clock, A]
val TimerTask = TaskR
```

Then encourage Cats Effect users to use this unless they need a more powerful / custom `R`.

**EDIT**: Anything else we should stick inside the Cats type alias aside from `Clock`?

### PR DESCRIPTION
**Pull request by gvolpe**
closes #694 

I wrote the `Timer[Task]` instance given the existing one. Is there a better way?

I also suggested adding the default `implicit val runtime` in a different trait but haven't done it. Let me know what you think.

### CODE REVIEW

File: interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala [hunk 1]
```diff
@@ -27,6 +27,22 @@ import scala.concurrent.duration.{ FiniteDuration, NANOSECONDS, TimeUnit }
 
 abstract class CatsPlatform extends CatsInstances {
   val console = interop.console.cats
+
+  object implicits {
+    implicit def taskTimer(implicit T: effect.Timer[ZIO[Clock, Throwable, ?]]): effect.Timer[Task] =
```
Comment (jdegoes) [inline interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala]: For improved performance, you could write this one:

```scala
implicit val taskTimer: effect.Timer[Task] = ???
```

Then over on the right hand side, use the `Clock.Live` methods directly to implement the functionality.

File: interop-cats/jvm/src/test/scala/scalaz/zio/interop/catzSpec.scala [hunk 2]
```diff
@@ -142,6 +142,16 @@ class catzSpec
...
 
+  object summoningRuntimeInstancesTest {
+    import cats.effect.{ Clock => CatzClock, _ }
+    import scalaz.zio.interop.catz.implicits._
+
+    ContextShift[Task]
+    ContextShift[TaskR[String, ?]]
+    CatzClock[Task]
+    Timer[Task]
+  }
```
Comment (jdegoes) [inline interop-cats/jvm/src/test/scala/scalaz/zio/interop/catzSpec.scala]: These are great. It occurs to me, users should be able to get them from `UIO`, too. Maybe worth adding a test here.

File: interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala (hunk 1 above)
Comment (jdegoes) [reply interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala]: If we want it to work for `UIO`, maybe we make it polymorphic in `E`:

```scala
implicit def taskTimer[E]: effect.Timer[IO[E, ?]] = ...
```

File: core/jvm/src/main/scala/scalaz/zio/App.scala [hunk 3]
```diff
@@ -39,6 +39,8 @@ package scalaz.zio
  */
 trait App extends DefaultRuntime {
 
+  implicit val runtime: Runtime[Environment] = this
```
Comment (jdegoes) [inline core/jvm/src/main/scala/scalaz/zio/App.scala]: I think maybe we should put this into the cats-interop package. That is, have an `App` that extends this app, inside cats-interop.

Then we could also provide other cats helpers there. What do you think?

File: interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala [hunk 4]
```diff
@@ -27,18 +27,34 @@ import scala.concurrent.duration.{ FiniteDuration, NANOSECONDS, TimeUnit }
 
 abstract class CatsPlatform extends CatsInstances {
   val console = interop.console.cats
+
+  object implicits {
+    implicit def ioTimer[E]: effect.Timer[IO[E, ?]] =
```
Comment (jdegoes) [inline interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala]: Looks great!

File: interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala [hunk 5]
```diff
@@ -27,18 +27,34 @@ import scala.concurrent.duration.{ FiniteDuration, NANOSECONDS, TimeUnit }
...
+      new effect.Timer[IO[E, ?]] {
+        override def clock: effect.Clock[IO[E, ?]] = new effect.Clock[IO[E, ?]] {
+          override def monotonic(unit: TimeUnit): IO[E, Long] =
+            zioClock.nanoTime.map(unit.convert(_, NANOSECONDS)).provide(Clock.Live)
```
Comment (jdegoes) [inline interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala]: You can just use `Clock.Live.nanoTime` (etc.) directly here and below, that way you avoid the whole `provide` thing. It is shorter and will be faster.

File: interop-cats/jvm/src/test/scala/scalaz/zio/interop/catzSpec.scala [hunk 6]
```diff
@@ -142,6 +142,20 @@ class catzSpec
...
+
+    ContextShift[Task]
+    ContextShift[TaskR[String, ?]]
+    CatzClock[Task]
+    Timer[Task]
+
+    ContextShift[UIO[?]]
+    CatzClock[UIO[?]]
+    Timer[UIO[?]]
+  }
```
Comment (jdegoes) [inline interop-cats/jvm/src/test/scala/scalaz/zio/interop/catzSpec.scala]: Superb!

File: core/jvm/src/main/scala/scalaz/zio/App.scala (hunk 3 above)
Comment (gvolpe) [reply core/jvm/src/main/scala/scalaz/zio/App.scala]: Yup, I think that'd be better.

Comment (gvolpe) [reply core/jvm/src/main/scala/scalaz/zio/App.scala]: Any preferences on naming?

Comment (jdegoes) [reply core/jvm/src/main/scala/scalaz/zio/App.scala]: `CatsApp`???

Comment (gvolpe) [reply core/jvm/src/main/scala/scalaz/zio/App.scala]: I was expecting `CatzApp` :smile: but that's good, I just pushed :)

**Comment by jdegoes** [comment]
giphy

**Pull request merged**